Sources/AsyncHTTPClient/ConnectionPool/HTTPConnectionPool+Factory.swift

```diff
@@ -434,7 +434,7 @@ extension HTTPConnectionPool.ConnectionFactory {
         }
         #endif
 
-        let sslServerHostname = self.key.connectionTarget.sslServerHostname
+        let sslServerHostname = self.key.serverNameIndicatorOverride ?? self.key.connectionTarget.sslServerHostname
```

Comment:
Should we hide this in a computed property on the key so as to centralise the knowledge of how to do this?

Reply:
Done. Actually found through this another place where I forgot to use the override, namely if we proxy:
https://github.com/swift-server/async-http-client/pull/675/commits/f296aa4cd01a074ff580791dcf7f6717f3c6a93a